tests/compiler/array_without_size/access_nested.leo

```diff
@@ -0,0 +1,10 @@
+/*
+namespace: Compile
+expectation: Pass
+input_file: input/dummy.in
+*/
+
+function main(y: bool) -> bool {
+    let d: [u8; (_, _)] = [[1,2,3,4], [1,2,3,4]];
```

Comment:
Feels like we should add DSTs to the language as a concept at this point and change the syntax to `[[u8]]`, like in Rust, as it is not really an array. This would also allow the syntax `_` in an expression context to denote an inference variable.

Reply:
@Centril DST meaning data summary types or something else?

As far as the `_` symbol for type inference I don't see a use case for that currently as just not typing something will trigger type inference. Seems like needless syntax unless I'm misunderstanding something.

As far as `[u8; (_, _)]` -> `[[u8]]` I know the big reason we didn't go a syntax like this was the current syntax seemed preferred for math heavy people that aren't programmers.

Reply:
DST meaning Dynamically Sized Type, e.g., slices in Rust. `[T]` being one example.

> As far as the `_` symbol for type inference I don't see a use case for that currently as just not typing something will trigger type inference. Seems like needless syntax unless I'm misunderstanding something.

Sometimes you want to specify the type but infer the length. For example, `let x: [u8; _] = make(); fn make3<T>() -> [T; 3] { ... }`. You're using return type polymorphism here.

> As far as `[u8; (_, _)]` -> `[[u8]]` I know the big reason we didn't go a syntax like this was the current syntax seemed preferred for math heavy people that aren't programmers.

Yeah. I think this is a mistake personally that mixes up concepts.

Reply:
Ah, I better understand the first point now.

For the latter, I know it was brought up by @damirka at a previous meeting but it was shot down after a discussion. Feel free to bring it up again though.